### PR TITLE
Add kitchen ci + fixes to pass tests

### PR DESCRIPTION
**Pull request by epcim**

### CODE REVIEW

File: .kitchen.openstack.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+platforms:
+    - name: ubuntu-14.04
+      driver:
+        username: root
+        image_ref: ubuntu-14-04-x64-1455869035
+        flavor_ref: m1.medium
+        #openstack_network_name: tcp_dev_pmi
+        network_ref:
+          - ci-net
```
Comment (fpytloun) [inline .kitchen.openstack.yml]: Should be probably also parametrized, the same for flavor and image above

File: .kitchen.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+
+platforms:
+  - name: bento/ubuntu-14.04
+#  - name: bento/ubuntu-16.04
+#  - name: bento/debian-jessie
+#  - name: bento/centos-7.1
+
+suites:
+
```
Comment (fpytloun) [inline .kitchen.yml:45]: Useless empty line here

File: .kitchen.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  - name: server
+    provisioner:
+      pillars-from-files:
+        postfix.sls: tests/pillar/server.sls
+      #pillars:
```
Comment (fpytloun) [inline .kitchen.yml]: I would remove useless comments, they are confusing without further explanation.

File: postfix/common.sls [hunk 4]
```diff
@@ -55,11 +55,24 @@ postfix_aliases:
   - require:
     - pkg: postfix_packages
 
+/var/lib/mailman/data:
```
Comment (fpytloun) [inline postfix/common.sls]: Here should be dependency on mailman packages (because they create list user/group)

File: postfix/map.jinja [hunk 5]
```diff
@@ -16,6 +16,10 @@
             '/etc/hosts', '/etc/localtime',
             '/etc/ssl/certs/ca-certificates.crt',
         ],
+        'ssl': {
+          'cert':  'AAAAA CERT AAAAA',
+          'chain': 'AAAA  CHAIN AAAA',
```
Comment (fpytloun) [inline postfix/map.jinja]: Don't add placeholders like this. Better ensure that proper `server.ssl` structure exists in code that needs it.

File: test/integration/relay/serverspec/relay_spec.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+end
+
+describe 'relay.sls, sasl_passwd' do
+
+  let(:sasl_passwd_file) { '/etc/postfix/sasl_passwd' }
+
+  it 'manages postfix sasl_passwd' do
+    expect(file(sasl_passwd_file).content).to match(/relay@cloudlab.cz:aes9aemiexohm7oohin4Amab9thahgha/)
```
Comment (fpytloun) [inline test/integration/relay/serverspec/relay_spec.rb:44]: Hard-coded user/password and more stuff, do we want it?

File: .kitchen.yml (hunk 2 above)
Comment (epcim) [inline .kitchen.yml:45]: will remove jessie, but all other are about to be enable as supported platforms. Thus will keep them so other devs may just uncoment.

File: test/integration/relay/serverspec/relay_spec.rb (hunk 6 above)
Comment (epcim) [inline test/integration/relay/serverspec/relay_spec.rb:44]: we should change our test data.. to example .com or .local etc. (but as of now they are everywhere). So answ. is we don't want but will keep that for later changes.

File: .kitchen.openstack.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          - <% ENV['OS_NETWORK_REF'] %>
+          <% else -%>
+          - ci-net
+          <% end -%>
+      # force update apt cache on the image
+      run_list:
+        - recipe[apt]
+      attributes:
+        apt:
+            compile_time_update: true
```
Comment (fpytloun) [inline .kitchen.openstack.yml]: long indent, the same above, yaml should have 2 spaces

File: postfix/map.jinja [hunk 8]
```diff
@@ -15,7 +15,7 @@
             '/etc/services', '/etc/resolv.conf', '/etc/nsswitch.conf',
             '/etc/hosts', '/etc/localtime',
             '/etc/ssl/certs/ca-certificates.crt',
-        ],
+        ]
```
Comment (fpytloun) [inline postfix/map.jinja]: Keep comma

**Pull request closed without merging**